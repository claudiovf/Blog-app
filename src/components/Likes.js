import React, { useState } from 'react'
import { useDispatch } from 'react-redux'
import { Button } from 'react-bootstrap'


const Likes = ({ blog, updateBlog }) => {
	const [blogLikes, setBlogLikes] = useState(blog.likes)
	const dispatch = useDispatch()

	const addLike = event => {
		event.preventDefault()

		dispatch(updateBlog(blog.id, {
			likes: blogLikes + 1,
		}))

		setBlogLikes(blogLikes + 1)

	}


	return (
		<div>
            Likes: {blogLikes}
			<Button variant="danger" name="likeButton" onClick={event => addLike(event)}>like</Button>
		</div>
	)
}

export default Likes