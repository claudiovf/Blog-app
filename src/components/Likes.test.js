import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { render, fireEvent } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Likes from './Likes'

describe('<Likes /> ', () => {
	let component
	const updateBlog = jest.fn()

	const blog = {
		title: 'Testing Likes component',
		author: 'Jest',
		url: 'www.bbc.com',
		likes: 30
	}

	const tree = renderer
		.create(<Likes blog={blog} updateBlog={updateBlog} />)
		.toJSON()

	expect(tree).toMatchSnapshot()

	beforeEach(() => {

		component = render(
			<Likes blog={blog} updateBlog={updateBlog} />
		)
	})

	test('receives the right info from LIKE button', () => {

		const likeButton = component.getByText('like')

		fireEvent.click(likeButton)
		fireEvent.click(likeButton)

		component.debug()
		expect(updateBlog.mock.calls).toHaveLength(2)

	})

})