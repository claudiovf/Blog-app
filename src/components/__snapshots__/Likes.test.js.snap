// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  Likes: 
  30
  <button
    onClick={[Function]}
  >
    like
  </button>
</div>
`;
